File: app/src/jvx/java/art/coded/wireframe/view/adapter/TabPagerAdapter.java
package art.coded.wireframe.view.adapter;

import android.content.Context;

import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import art.coded.wireframe.R;
import art.coded.wireframe.view.fragment.TabFragment;

/**
 * A [FragmentPagerAdapter] that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
public class TabPagerAdapter extends FragmentPagerAdapter {

    private static final String LOG_TAG = TabPagerAdapter.class.getSimpleName();

    @StringRes private static final int[] TAB_TITLES = new int[]{R.string.tab_text_1, R.string.tab_text_2};
    private final Context mContext;

    public TabPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        mContext = context;
    }

    @Override public Fragment getItem(int position) {
        // getItem is called to instantiate the fragment for the given page.
        // Return a TabFragment (defined as a static inner class below).
        return TabFragment.newInstance(position + 1);
    }

    @Nullable @Override public CharSequence getPageTitle(int position) {
        return mContext.getResources().getString(TAB_TITLES[position]);
    }

    @Override public int getCount() {
        // Show 2 total pages.
        return 2;
    }
}

File: app/src/ktx/java/art/coded/wireframe/view/adapter/TabPagerAdapter.kt
package art.coded.wireframe.view.adapter

import art.coded.wireframe.R
import androidx.fragment.app.FragmentPagerAdapter
import art.coded.wireframe.view.fragment.TabFragment
import androidx.annotation.StringRes
import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

private val LOG_TAG = TabPagerAdapter::class.java.simpleName
@StringRes private val TAB_TITLES = intArrayOf(R.string.tab_text_1, R.string.tab_text_2)

/**
 * A [FragmentPagerAdapter] that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
class TabPagerAdapter(private val mContext: Context, fm: FragmentManager?) : FragmentPagerAdapter(
    fm!!
) {
    override fun getItem(position: Int): Fragment {
        // getItem is called to instantiate the fragment for the given page.
        // Return a TabFragment (defined as a static inner class below).
        return TabFragment.newInstance(position + 1)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mContext.resources.getString(TAB_TITLES[position])
    }

    override fun getCount(): Int {
        // Show 2 total pages.
        return 2
    }
}